thriftbp/headers.go

```diff
@@ -44,3 +45,17 @@ func AddClientHeader(ctx context.Context, key, value string) context.Context {
 	headers = append(headers, key)
 	return thrift.SetWriteHeaderList(ctx, headers)
 }
+
+// header gets the value of a thrift header by key
+//
+// If the value is not present we fall back to a case-insensitive check
```

Comment:
s/fall back to a case-insensitive check/fall back to all lowercase check to workaround a bug in envoy/